src/Ide/Plugin/Eval.hs

```diff
@@ -309,6 +303,54 @@ done, we want to switch back to GhcSessionDeps:
 
     return (WorkspaceApplyEdit, ApplyWorkspaceEditParams workspaceEdits)
 
+evalGhciLikeCmd :: Text -> Text -> Ghc (Maybe Text)
+evalGhciLikeCmd cmd arg = do
+  df <- getSessionDynFlags
+  let tppr = T.pack . showSDoc df . ppr
+  case cmd of
+    "kind" -> do
+      let input = T.strip arg
+      (_, kind) <- typeKind False $ T.unpack input
+      pure $ Just $ "-- " <> input <> " :: " <> tppr kind <> "\n"
+    "kind!" -> do
+      let input = T.strip arg
+      (ty, kind) <- typeKind True $ T.unpack input
+      pure
+        $ Just
+        $ T.unlines 
+        $ map ("-- " <>)
+        [ input <> " :: " <> tppr kind
+        , "= " <> tppr ty
+        ]  
+    "type" -> do
+      let (emod, expr) = parseExprMode arg
+      ty <- exprType emod $ T.unpack expr
+      pure $ Just $ 
+        "-- " <> expr <> " :: " <> tppr ty <> "\n"
+    _ -> E.throw $ GhciLikeCmdNotImplemented cmd arg
+
+parseExprMode :: Text -> (TcRnExprMode, T.Text)
+parseExprMode rawArg =
+  case T.break isSpace rawArg of
+    ("+v", rest) -> (TM_NoInst, T.strip rest)
+    ("+d", rest) -> (TM_Default, T.strip rest)
+    _ -> (TM_Inst, rawArg)
+
+data GhciLikeCmdException = GhciLikeCmdNotImplemented Text Text
```

Comment:
I would prefer if the fields were named, so that it's obvious what they mean

Reply:
Exactly. I've just added field names!